NetKet/Graph/custom_graph.hpp

```diff
@@ -44,73 +67,175 @@ class CustomGraph : public AbstractGraph {
   bool is_connected_;
 
  public:
-  // Json constructor
-  explicit CustomGraph(const json &pars) { Init(pars); }
 
-  void Init(const json &pars) {
-    // Try to construct from explicit graph definition
-    if (FieldExists(pars, "Graph")) {
-      if (FieldExists(pars["Graph"], "AdjacencyList")) {
-        adjlist_ =
-            pars["Graph"]["AdjacencyList"].get<std::vector<std::vector<int>>>();
-      }
-      if (FieldExists(pars["Graph"], "Edges")) {
-        std::vector<std::vector<int>> edges =
-            pars["Graph"]["Edges"].get<std::vector<std::vector<int>>>();
-        AdjacencyListFromEdges(edges);
-      }
-      if (FieldExists(pars["Graph"], "Size")) {
-        assert(pars["Graph"]["Size"] > 0);
-        adjlist_.resize(pars["Graph"]["Size"]);
+#if 0 // TODO(twesterhout): To be removed
+  explicit CustomGraph(
+      int size = 0,
+      std::vector<std::vector<int>> adjl = std::vector<std::vector<int>>(),
+      std::vector<std::vector<int>> edges = std::vector<std::vector<int>>(),
+      std::vector<std::vector<int>> automorphisms =
+          std::vector<std::vector<int>>(),
+      std::vector<std::vector<int>> edgecolors =
+          std::vector<std::vector<int>>(),
+      bool isbipartite = false)
+      : adjlist_(adjl), isbipartite_(isbipartite) {
+    Init(size, edges, automorphisms, edgecolors);
+  }
+#endif
+
+  CustomGraph(std::vector<Edge> edges, ColorMap colors = ColorMap(),
```

Comment:
It seems we cannot construct this anymore just passing CustomGraph(size=10) say. 
The only unit tests in python we have are failing. `Test/Graph/test_graph.py`.

Reply:
We need to enable these tests at least, otherwise it's going to be a disaster in the following with all these changes

Reply:
I have partially complete python tests for Hilbert and Operator, do you want me to push those now and then finish them up as we go along?

Reply:
@gcarleo Indeed we can't. I don't think such graphs make much sense though or do they?